.gitlab-ci.yml

```diff
@@ -1,28 +1,23 @@
-image: python:3.7
+# GitLab Dependency Proxy to get rid of Docker hub pull limit issue
+# https://docs.gitlab.com/ee/user/packages/dependency_proxy/#authenticate-within-cicd
+image: ${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/python:3.7
 
 # GitLab CI uses the services keyword to define what docker containers should be linked with our base image.
 # Since we are using docker as our image we'll be using docker:dind i.e. Docker-In-Docker-Image
 services:
-  - name: docker:19.03.12-dind
+  - name: ${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/docker:dind
 
 variables:
-  # Environemt Variable for docker:dind service explaining to use overlay2 as supporting driver for docker
+  # This fixes "error during connect: Post http://docker:2375/v1.40/auth: dial tcp: lookup docker on 192.168.65.5:53: no such host"
```

Comment:
It's not clear to me from this comment what's fixing this error.  Is it the DOCKER_HOST env variable?  Is it the fact that the port number is 2375 rather than 2376?